### DISCUSSION TITLE
no data on ebus for Vaillant VRC700 (ERR: wrong symbol received)

### DISCUSSION THREAD

**Discussion by bertvandevoorde** [john30/ebusd#750] [Q&A]
Dear all, Dear @john30 ,

I've been struggling for two days to get reliable data on the ebus of my Vaillant Ecotec system. In any case I have to make sure that my ebus usb coupler device is running before I start the heating system. If I don't do this, then it won't detect all of the devices during the scan. It will even take half an our or so before it detects the thermostat VRC700. It will then show up as detected in "ebusctl info" but it will not load the 15.700.csv configuration file. 

I'll try to give as much info as possible so please bare with me:

I'm on the following system:
Vaillant Ecotec boiler Ecotec plus, two Thermostats VRC700 and VR91 + central pump steering VR70. 
Ebus USB coupler device: (https://esera.de/en/Products/eBus/135/eBus-Coupler-USB) 
Raspberry pi 2 (armv7) running ebusd 22.4 +  ebusd-configuration 2.1.x (de)
- I have set "setserial /dev/ttyUSB0 low_latency". 
- I use the following foreground command to run ebusd: ebusd -f -d /dev/ttyUSB0 --scanconfig=15 -c /etc/ebusd/ --mqtthost=192.168.1.100 --mqttport=1883 --mqttuser=user --mqttpass=pass --pollinterval=180 --latency=20000 --acquiretimeout=15000
--> notice how I don't specify --scanconfig=full but rather focus on the address of the VRC700.

I have a feeling that I don't have a lot of margin here, the slightest difference in any of the parameters and this impacts scanning behaviour and would make it so that certain csv files are not loaded. With the exact configuration above, I reach a point where my infrastructure is detected and it would load at least 3 csv files. See below:

address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=BAI00;SW=0105;HW=7803", loaded "vaillant/bai.308523.inc", "vaillant/08.bai.csv"
address 10: master #2
address 15: slave #2, scanned "MF=Vaillant;ID=70000;SW=0614;HW=6903", loaded "vaillant/15.700.csv"
address 30: master #3
address 31: master #8, ebusd
address 35: slave #3, scanned "MF=Vaillant;ID=VR_91;SW=0415;HW=4803"
address 36: slave #8, ebusd
address 52: slave, scanned "MF=Vaillant;ID=VR_70;SW=0109;HW=2903", loaded "vaillant/52.vr_70.csv"

I'm able to read a couple of simple counters like "outsidetemp" and I have data for the boiler (BAI00) like the watertemperature.  I also  have all of this in my MQTT broker. The thing is that anything coming from the VRC700 (15.700.csv) remains unreadable. "find -v" will show "no data stored. If I try to read something: "read Hc1PumpStatus" then I get "ERR: wrong symbol received" every time. It's like the system doesn't know how to make sense from what is coming in from the VRC700.

The logging shows resolved data for all the components but then regularly throws messages like this
2022-12-10 14:47:35.357 [update notice] received unknown MS cmd: 3015b5240a06010a01070000007842 / 02000a
I guess these come from the VRC700

Actually I feel like something is acting very sensitive. The "ERR: wrong symbol received" error troubles me a lot and it is also this error which is very often thrown if I start the USB coupler while the heating system is already running. I then receive this errors for all addresses and the ebus service never reaches a point where scanning finishes. Even if scanning does finish for one device, it would never load the csv.

Thank you for taking the time to read this. Thanks for your excellent effort maintaining this ebus software.
Would you (@john30) or anyone have any suggestions for me?

**Comment by mwildbolz** [comment]
Maybe your adapter isn't able to correctly write to the bus.
I had the same problem recently with an ESERA eBus USB adapter, other users also report this.

Maybe you could also use the vr921 to poll the data and read it [like here](https://github.com/john30/ebusd/discussions/735)

**Reply by bertvandevoorde** [reply]
Thank you. You're probably right. The esera drove me mad this weekend :). This referred post is very interesting. I would need to buy the vr921 though, I don't have it yet. I guess I should also opt in for the ebus adapter 3.

**Reply by bertvandevoorde** [reply]
Update on the topic: I'm actually running read only now with a copy of the relevant csv files. I removed --scanconfig. I'm back to the issue now where the messages get decoded for the BAI and VR70, but not for the thermostat VRC700. I'm still seeing lines like this:

2022-12-11 21:26:10.145 [update notice] received unknown MS cmd: 1008b5110100 / 085c0211001f000080
2022-12-11 21:27:10.403 [update notice] received unknown MS cmd: 1008b5110100 / 085a0211001f000080

I suspect these are coming from the VRC700 (15.700.csv) and do not get decoded. This is unfortunate because there's certainly interesting information there. What could be the reason? Perhaps the received messages happen to not be in the 15.700.csv and I'm not actually getting anything that is. Although that would be weird. Puzzled.

ebusctl info resolves all equipment as you can see below:

root@unipi:/etc/ebusd# ebusctl info
version: ebusd 22.4.v22.4
update check: OK
device: /dev/serial/by-id/usb-ESERA_GmbH_eBus_Coupler_12001_AL7BLZ46-if00-port0
signal: acquired
[.... removed for brevity....]
address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=BAI00;SW=0105;HW=7803"
address 10: master #2
address 15: slave #2, scanned "MF=Vaillant;ID=70000;SW=0614;HW=6903"
address 30: master #3
address 31: master #8, ebusd
address 35: slave #3, scanned "MF=Vaillant;ID=VR_91;SW=0415;HW=4803"
address 36: slave #8, ebusd
address 52: slave, scanned "MF=Vaillant;ID=VR_70;SW=0109;HW=2903"

The files in /etc/ebusd and /etc/ebusd/vaillant

root@unipi:/etc/ebusd# find .
.
./broadcast.csv
./mqtt-hassio.cfg
./memory.csv
./mqtt-integration.cfg
./vaillant
./vaillant/bai.0010006101.inc
./vaillant/bai.0010002465.inc
./vaillant/bai.0010006341.inc
./vaillant/bai.0010008045.inc
./vaillant/broadcast.csv
./vaillant/bai.0010010674.inc
./vaillant/bai.0010003886.inc
./vaillant/errors.inc
./vaillant/bai.0010005400.inc
./vaillant/bai.308523.inc
./vaillant/bai.0020066007.inc
./vaillant/bai.0010015600.inc
./vaillant/scan.csv
./vaillant/bai.0010004121.inc
./vaillant/hcmode.inc
./vaillant/08.bai.csv
./vaillant/bai.0010003857.inc
./vaillant/bai.0010021961.inc
./vaillant/52.vr_70.csv
./vaillant/bai.0010004150.inc
./vaillant/_templates.csv
./vaillant/service.inc
./vaillant/hwcmode.inc
./vaillant/bai.0010007508.inc
./vaillant/15.700.csv
./vaillant/general.csv
./vaillant/bai.0010002315.inc
./_templates.csv
./knx.cfg

ebusd --checkconfig looks good at first glance.
root@unipi:/etc/ebusd# ebusd --checkconfig -c /etc/ebusd
2022-12-11 21:35:01.686 [main notice] ebusd 22.4.v22.4 performing configuration check...
2022-12-11 21:35:03.012 [main notice] found messages: 5038 (19 conditional on 33 conditions, 0 poll, 10 update)

the --dumpconfig does contain lines for the 700. Snippet below

r,700,HwcHolidayStartPeriod,HwcHolidayStartPeriod,,15,b524,020001000900,,s,IGN:4,,,,hfrom,s,HDA:3,,,start date of holidays
w,700,HwcHolidayStartPeriod,HwcHolidayStartPeriod,,15,b524,020101000900,hfrom,m,HDA:3,,,start date of holidays
r,700,HwcLockTime,lockTimeHwc,,15,b524,020000001800,,s,IGN:4,,,,minutes2,s,UIN,,min,Sperrzeit WW-Bedarf
w,700,HwcLockTime,lockTimeHwc,,15,b524,020100001800,minutes2,m,UIN,,min,Sperrzeit WW-Bedarf
r,700,HwcMaxFlowTempDesired,HwcMaxFlowTempDesired,,15,b524,020000004600,,s,IGN:4,,,,tempv,s,EXP,,°C,maximum flow temperature setpoint of Hwc
w,700,HwcMaxFlowTempDesired,HwcMaxFlowTempDesired,,15,b524,020100004600,tempv,m,EXP,,°C,maximum flow temperature setpoint of Hwc
r,700,HwcOpMode,HwcOpMode,,15,b524,020001000300,,s,IGN:4,,,,opmode,s,UIN,0=off;1=auto;2=day;3=night,,operation mode of Hwc
w,700,HwcOpMode,HwcOpMode,,15,b524,020101000300,opmode,m,UIN,0=off;1=auto;2=day;3=night,,operation mode of Hwc
r,700,HwcParallelLoading,Parallele Ladung Warmwasserkreis,,15,b524,020000000a00,,s,IGN:4,,,,onoff,s,UCH,0=off;1=on,,Heizbetrieb und Speicherladung parallel
w,700,HwcParallelLoading,Parallele Ladung Warmwasserkreis,,15,b524,020100000a00,onoff,m,UCH,0=off;1=on,,Heizbetrieb und Speicherladung parallel
